### PR TITLE
refactor(Auth): Changed the Auth plugins user agent to use Amplify

### PR DESCRIPTION
**Pull request by royjit**
Adds AmplifyServiceConfiguration to Auth category. Now the user agent shows the amplify-io string when making Cognito service calls.

Rev 2
Updated all aws sdk version to 2.13.5

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/Core/AWSPluginsCore/ServiceConfiguration/AmplifyAWSServiceConfiguration.swift [hunk 1]
```diff
@@ -36,4 +36,8 @@ public class AmplifyAWSServiceConfiguration: AWSServiceConfiguration {
                          credentialsProvider: AWSCredentialsProvider) {
         super.init(region: regionType, credentialsProvider: credentialsProvider)
     }
+
+    public init(region regionType: AWSRegionType) {
+        super.init(region: regionType, credentialsProvider: nil)
+    }
```
Comment (palpatim) [inline AmplifyPlugins/Core/AWSPluginsCore/ServiceConfiguration/AmplifyAWSServiceConfiguration.swift:42]: Do we need this? We already have an initializer that takes a `nil`, as used on L32 above.

File: AmplifyPlugins/Auth/Podfile.lock [hunk 2]
```diff
@@ -26,7 +26,7 @@ PODS:
   - CwlCatchException (1.0.2)
   - CwlPreconditionTesting (1.1.1):
     - CwlCatchException
-  - SwiftFormat/CLI (0.44.13)
+  - SwiftFormat/CLI (0.44.14)
```
Comment (palpatim) [inline AmplifyPlugins/Auth/Podfile.lock:29]: Don't we also need to declare a dependency on AWSMobileClient `~> 2.13.5`?

File: AmplifyPlugins/Auth/AWSCognitoAuthPlugin/AWSCognitoAuthPlugin+Configure.swift [hunk 3]
```diff
@@ -55,6 +62,35 @@ extension AWSCognitoAuthPlugin {
         }
     }
 
+    func identityPoolServiceConfiguration(from authConfiguration: JSONValue) throws -> AmplifyAWSServiceConfiguration {
+        let regionKeyPath = "CredentialsProvider.CognitoIdentity.Default.Region"
+        guard case .string(let regionString) = authConfiguration.value(at: regionKeyPath) else {
+            let amplifyError = AuthError.configuration(
+                "Error configuring \(String(describing: self))",
+                """
+                Couldnot read Cognito identity pool information from the configuration
```
[the remark this answers is not in the document]
Comment (royjit) [reply AmplifyPlugins/Auth/AWSCognitoAuthPlugin/AWSCognitoAuthPlugin+Configure.swift]: 1. Fixed
2. Updated the message

File: AmplifyPlugins/Auth/Podfile.lock (hunk 2 above)
Comment (royjit) [reply AmplifyPlugins/Auth/Podfile.lock:29]: Updated

File: AmplifyPlugins/Core/AWSPluginsCore/ServiceConfiguration/AmplifyAWSServiceConfiguration.swift (hunk 1 above)
Comment (royjit) [reply AmplifyPlugins/Core/AWSPluginsCore/ServiceConfiguration/AmplifyAWSServiceConfiguration.swift:42]: That initializer is overriden by the initilizer in line L35 that take a nonnull `credentialsProvider`. So I cannot init this class outside like this: `AmplifyAWSServiceConfiguration(region: region, credentialsProvider: nil)`

**Pull request merged**